public/js/utils/utils.js

```diff
@@ -62,18 +61,29 @@ function endTruncateStr(str: any, size: any) {
   return str;
 }
 
-function workerTask(worker: any, message: any) {
-  let deferred = defer();
-  worker.postMessage(message);
-  worker.onmessage = function(result) {
-    if (result.data && result.data.error) {
-      deferred.reject(result.data.error);
-    }
-
-    deferred.resolve(result.data);
+let msgId = 1;
+function workerTask(worker: any, method: string) {
+  return function(...args) {
+    return new Promise((resolve, reject) => {
+      const id = msgId++;
+      worker.postMessage({ id, method, args });
+
+      const listener = ({ data: result }) => {
+        if (result.id !== id) {
+          return;
+        }
+
+        worker.removeEventListener("message", listener);
+        if (result.error) {
+          reject(result.error);
+        } else {
+          resolve(result.response);
+        }
+      };
+
+      worker.addEventListener("message", listener);
+    });
```

Comment:
👍 

I believe this is similar to how the old source map worker communicated. Out of curiosity, was the motivation for the IDs the `async` functions in the worker? I believe, the old worker API was internally synchronous.